### PR TITLE
[quant] Refactor test files

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36964 [quant] Refactor test files**

Summary:
Rename and restructure quantization related tests
https://github.com/pytorch/pytorch/issues/31625

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21192509](https://our.internmc.facebook.com/intern/diff/D21192509)

### CODE REVIEW

File: test/quantization/test_backward_compatibility.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # JIT tests use double as the default dtype
 torch.set_default_dtype(torch.double)
 
-class TestSerialization(TestCase):
+class TestBC(TestCase):
```
Comment (vkuzo) [inline test/quantization/test_backward_compatibility.py:21]: what does BC stand for?

**Review by vkuzo — APPROVED** [review]
lg, one nit question inline

File: test/quantization/test_backward_compatibility.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_backward_compatibility.py:21]: backward compatibility, this is too long so I abbreviated it to BC

**Review by raghuramank100 — APPROVED** [review]

File: test/quantization/test_backward_compatibility.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_backward_compatibility.py]: turns out I need to change this back because the test name is in the serialized file

**Comment by z-a-f** [comment]
BTW, this diff removed the ability to run the tests directly

**Pull request closed without merging**